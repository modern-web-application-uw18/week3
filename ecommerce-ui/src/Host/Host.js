import React, {Component} from 'react';
import PropTypes from 'prop-types';

class Host extends Component {
    render() {

        const isSuperiorHost = () => {
            if ( this.props.host.isSuperhost ) {
                return ( <span> {this.props.host.name} &middot; Superhost </span> );
            }
            else {
                return ( <span> {this.props.host.name} </span> );
            }

        }

        return(
            <span> {isSuperiorHost()} </span>
        )
    };
}

Host.propTypes = {
    name: PropTypes.string.isRequired,
    isSuperhost: PropTypes.bool
}

Host.defaultProps = {
    name: "No name given",
    isSuperhost: false
}

export default Host;