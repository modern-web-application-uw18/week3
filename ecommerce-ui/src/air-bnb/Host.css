* {
    box-sizing: border-box;
}

.Host {
    font-weight: 500;
    font-size: 0.85em;
}