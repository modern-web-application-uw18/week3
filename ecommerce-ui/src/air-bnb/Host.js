import React, { Component } from 'react';
import PropTypes from 'prop-types';
import Rating from './Rating';
import './Host.css';

class Host extends Component {

    render() {
        
        const {
            name,
            isSuperhost
        } = this.props.host;

        const isSuper = () => {
            if (isSuperhost) {
                return (
                    <span> &middot; Superhost </span>
                )
            } else {
                return null;
            }
        }

        return (
            <div className="Host">{name} {isSuper()} &middot; <Rating rating={this.props.rating}/></div>
        );
    }
}

Host.propTypes = {
    name: PropTypes.string.isRequired,
    isSuperhost: PropTypes.bool,
    rating: PropTypes.instanceOf(Object)
};

Host.defaultProps = {
    name: 'Anonymous',
    isSuperhost: false
};

export default Host;