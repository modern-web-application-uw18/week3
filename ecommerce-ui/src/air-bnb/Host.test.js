import React from 'react';
import ReactDOM from 'react-dom';
import TestRenderer from 'react-test-renderer';
import Host from './Host';

describe('Host', () => {
    it('renders without crashing', () => {
        const rating = {stars: 4, reviews: 123};
        const host = {name: "George Bush", isSuperhost: true};
        const div = document.createElement('div');
        ReactDOM.render(<Host host={host} rating={rating} />, div);
        ReactDOM.unmountComponentAtNode(div);
    });

    it('should render with host name, superhost, stars and reviews', () => {
        const rating = {stars: 4.5, reviews: 313};
        const host = {name: "Steve Powers", isSuperhost: false};
        const component = TestRenderer.create(
            <Host host={host} rating={rating}/>);
            expect(component.toJSON()).toMatchSnapshot();
    });
});