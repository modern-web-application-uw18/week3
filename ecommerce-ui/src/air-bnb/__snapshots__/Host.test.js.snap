// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Host should render with host name, superhost, stars and reviews 1`] = `
<div
  className="Host"
>
  Steve Powers
   
   · 
  <span>
    313
     reviews ·
    <span
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      style={
        Object {
          "direction": "ltr",
          "display": "inline-block",
        }
      }
    >
      <span
        onClick={[Function]}
        onMouseMove={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        style={
          Object {
            "cursor": "pointer",
            "display": "inline-block",
            "position": "relative",
          }
        }
      >
        <img
          alt=""
          src="gray-star.png"
        />
        <span
          style={
            Object {
              "display": "inline-block",
              "left": 0,
              "overflow": "hidden",
              "position": "absolute",
              "top": 0,
              "width": "100%",
            }
          }
        >
          <img
            alt=""
            src="green-star.png"
          />
        </span>
      </span>
      <span
        onClick={[Function]}
        onMouseMove={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        style={
          Object {
            "cursor": "pointer",
            "display": "inline-block",
            "position": "relative",
          }
        }
      >
        <img
          alt=""
          src="gray-star.png"
        />
        <span
          style={
            Object {
              "display": "inline-block",
              "left": 0,
              "overflow": "hidden",
              "position": "absolute",
              "top": 0,
              "width": "100%",
            }
          }
        >
          <img
            alt=""
            src="green-star.png"
          />
        </span>
      </span>
      <span
        onClick={[Function]}
        onMouseMove={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        style={
          Object {
            "cursor": "pointer",
            "display": "inline-block",
            "position": "relative",
          }
        }
      >
        <img
          alt=""
          src="gray-star.png"
        />
        <span
          style={
            Object {
              "display": "inline-block",
              "left": 0,
              "overflow": "hidden",
              "position": "absolute",
              "top": 0,
              "width": "100%",
            }
          }
        >
          <img
            alt=""
            src="green-star.png"
          />
        </span>
      </span>
      <span
        onClick={[Function]}
        onMouseMove={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        style={
          Object {
            "cursor": "pointer",
            "display": "inline-block",
            "position": "relative",
          }
        }
      >
        <img
          alt=""
          src="gray-star.png"
        />
        <span
          style={
            Object {
              "display": "inline-block",
              "left": 0,
              "overflow": "hidden",
              "position": "absolute",
              "top": 0,
              "width": "100%",
            }
          }
        >
          <img
            alt=""
            src="green-star.png"
          />
        </span>
      </span>
      <span
        onClick={[Function]}
        onMouseMove={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        style={
          Object {
            "cursor": "pointer",
            "display": "inline-block",
            "position": "relative",
          }
        }
      >
        <img
          alt=""
          src="gray-star.png"
        />
        <span
          style={
            Object {
              "display": "inline-block",
              "left": 0,
              "overflow": "hidden",
              "position": "absolute",
              "top": 0,
              "width": "50%",
            }
          }
        >
          <img
            alt=""
            src="green-star.png"
          />
        </span>
      </span>
    </span>
  </span>
</div>
`;
